Pending-Name-21/frontend-library
# Implementation of input intermediate

# Task-207 Implementation of Input intermediate

https://tree.taiga.io/project/joseluis-teran-coffeetime/task/207

## Description

- #### What did I do?
Following the[ architecture update](https://github.com/Pending-Name-21/arquitecture/pull/17), the client socket component has been migrated to the bridge.

Implemented Events un-marshalling 

#### How did I do it?
- Added implementation documentation.

- Updated CI to ignore CoffeeTime (machine-generated code).

- Implemented event buffers:
  - Receiver now populates a thread-safe LIFO collection with deserialized client data.
  - Refactored Receiver to handle ByteBuffer, with SocketServer managing the socket.

- Added FlatBuffer communication protocol:
  - FlatBuffer is an efficient serialization library.
  - KeyboardEvents are generated on user input and deserialized using FlatBuffer.

- Created SocketServer:
  - A Runnable that processes client data with the Receiver feeding the buffers.


#### Why did I do it?
  To improve maintainability, enhance functionality with new features, and streamline exception handling and testing.

## Testing Instructions
Remove the Disabled line from bridge/src/test/java/com/bridge/ipc/SocketServerTest.java
Execute:
`$ ./gradlew test --tests com.bridge.ipc.SocketServerTest#startServer`

Open another terminal, and checkout the branch [spike-xx/serialization](https://github.com/Pending-Name-21/screen/tree/spike-xx/serialization) (It's from the screen repo)
Then execute 
`$ cargo --quiet test send_flattbuf`

You should see something like this.
![image](https://github.com/Pending-Name-21/frontend-library/assets/64939916/8e68111f-4421-46c8-83b1-0d382c473d61)



## Type of Change

Put an `x` in all the boxes that apply:


## Pull Request Checklist

- [x] My commit messages are detailed.
- [ ] No existing features have been broken without good reason.
